common/eth2/src/lighthouse.rs

```diff
@@ -71,16 +73,118 @@ pub struct ValidatorInclusionData {
     pub is_previous_epoch_head_attester: bool,
 }
 
+#[cfg(target_os = "macos")]
+use psutil::process::Process;
 #[cfg(target_os = "linux")]
-use {procinfo::pid, psutil::process::Process};
+use psutil::process::Process;
+
+/// Reports information about the system the Lighthouse instance is running on.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct System {
+    pub health: Health,
+    pub drives: Vec<Drive>,
+}
+
+impl System {
+    pub fn observe() -> Result<Self, String> {
+        Ok(Self {
+            health: Health::observe()?,
+            drives: Drive::observe()?,
+        })
+    }
+}
+
+/// Reports information about a drive on the system the Lighthouse instance is running on.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Drive {
+    /// The filesystem name.
+    pub filesystem: String,
+    /// The amount of disk space available on the filesystem.
+    pub avail: u64,
+    /// The amount of disk space used on the filesystem. Equivalent to `total-avail`.
+    pub used: u64,
+    /// The percentage of disk space used on the filesystem. Equivalent to `(total-avail) / total`.
+    pub used_pct: u64,
+    /// The total amount of disk space on the filesystem.
+    pub total: u64,
+    /// The filesystem mount point.
+    pub mounted_on: String,
+}
+
+impl Drive {
+    pub fn observe() -> Result<Vec<Self>, String> {
+        let system = SystemStat::new();
+        Ok(system
+            .mounts()
+            .expect("Could not find mounts.")
+            .into_iter()
+            // filter out drives with zero total disk space
+            .filter(|drive| drive.total.as_u64() != 0)
+            .map(|drive| Drive {
+                filesystem: drive.fs_mounted_from,
+                avail: drive.avail.as_u64(),
+                used: drive.total.as_u64() - drive.avail.as_u64(),
+                used_pct: (((drive.total.0 as f64 - drive.avail.0 as f64) / drive.total.0 as f64)
```

Comment:
I recommend using saturating arith for this subtraction.

Also, I see the divide-by-zero is protected by an earlier `filter`. I would recommend using safe math here or adding a comment to that `filter` saying that it protects against a panic.